src/probability/independence.lean

```diff
@@ -525,6 +543,94 @@ begin
   { exact ⟨ψ ⁻¹' B, hψ hB, set.preimage_comp.symm⟩ }
 end
 
+/-- If `f` is a family of mutually independent random variables (`Indep_fun m f μ`) and `S,T` are
+two disjoint index sets, then the finite product of `f` over `S` is independent of the finite
+product of `f` over `T`. -/
+lemma Indep_fun.indep_fun_finset [is_probability_measure μ]
+  {ι : Type*} {β : ι → Type*} {m : Π i, measurable_space (β i)}
+  {f : Π i, α → β i} (S T : finset ι) (hST : disjoint S T) (hf_Indep : Indep_fun m f μ)
```

Comment:
You take `S` and `T` to be finite. Is that really relevant here? Also, the word "product" in the docstring is a little bit misleading: I was thinking of some multiplication...

Reply:
I doubt it is really relevant mathematically, but some results in `measure_theory/constructions/pi.lean` require fintypes. I'll check whether they actually need it.

Reply:
It looks like everything in the constructions.pi file relies on fintypes for the index sets, because it uses repeatedly that `ι → ℕ` is encodable.

Reply:
Yes, tuple is much better. Let's go with finsets for now, and generalize if/when needed.